### ISSUE TITLE
RESTORE on a table complains about another index being open

### ISSUE THREAD

**Issue by benjaminrigaud** [crate/crate#3316]
Hello Crate team,

When I do (on Crate 0.54.5):
```
RESTORE SNAPSHOT snapshots.snapshot_prod_2016_03_01 TABLE events_2015_10 WITH (wait_for_completion=true);
```
I get:
```
SQLActionException[[snapshots:snapshot_prod_2016_03_01] cannot restore index [events_2015_07] because it's open]
```

The snapshot contains all the tables.

Any idea?

**Comment by joemoe** [comment]
hi @benjaminrigaud, are you doing anything else with crate while executing the restore query?

**Comment by benjaminrigaud** [comment]
There may be some queries running on other indices.

The strange part is events_2015_07 preventing a restore on events_2015_10.

**Comment by joemoe** [comment]
ok. thanks we'll look at it.

**Comment by mfussenegger** [comment]
Can you provide the `CREATE REPOSITORY` and `CREATE SNAPSHOTS` commands that you used to create the repo and snapshot?

**Comment by benjaminrigaud** [comment]
Sure.

`CREATE REPOSITORY snapshots TYPE S3 WITH (bucket='....', region='eu-central-1', base_path='snapshots', access_key='...', secret_key='...');`

`CREATE SNAPSHOT snapshots.snapshot_prod... ALL WITH (wait_for_completion=true);`

**Comment by mfussenegger** [comment]
I'm not able to reproduce the error (unless I do restore ALL - which is to be expected).

Do you have any partitioned tables in your cluster or within the snapshot?

**Comment by benjaminrigaud** [comment]
No.

**Comment by mfussenegger** [comment]
Since you've written in slack that you'll do some additional tests I'll wait here for the results.

If you can come up with steps to reproduce this it would be great.

Unfortunately there is also almost no logging in the restore code which would provide any insight. But I could provide a custom built with more logging included if that would be an option for you

**Comment by benjaminrigaud** [comment]
It would be an option :) As soon as I can reproduce this: I don't have incoming RESTORE to do on the production cluster. I'll have to setup a test cluster.

**Comment by tellezb** [comment]
I just reproduced this on my (single-node) system.

`CREATE REPOSITORY crate_snapshots TYPE fs WITH (location='/crate_snapshots', compress=true);`

`CREATE SNAPSHOT crate_snapshots.snapshot1 ALL WITH (wait_for_completion=true, ignore_unavailable=true);`

Which created the following snapshot:

> repository: crate_snapshots
> name: snapshot1
> STATE: SUCCESS
> concrete_indices[".partitioned.events.044j4c1h6oo36chk",".partitioned.events.044j4c1h6oo36chi","tweets",".partitioned.events.044j4c1h6oo36chj"]

After dropping the tweets table and then attempting to restore, I get the error.

`RESTORE SNAPSHOT crate_snapshots.snapshot1 TABLE tweets WITH (wait_for_completion=true)`

> SQLActionException[[crate_snapshots:snapshot1] cannot restore index [.partitioned.events.044j4c1h6oo36chk] because it's open]

After dropping the .partitioned.events.044j4c1h6oo36chk partition from my events table, I attempted the tweets restore again using the same RESTORE SNAPSHOT script. **The dropped partition in my events table was restored instead of the tweets table like I requested.**

Here's the schema for my events table:

```
CREATE TABLE IF NOT EXISTS "doc"."events" (
   "id" LONG,
   "reseller_id" INTEGER,
   "day" STRING, 
   "value" FLOAT,
   PRIMARY KEY ("id", "reseller_id", "day")
)
CLUSTERED BY ("reseller_id") INTO 500 SHARDS PARTITIONED BY ("day")
WITH (number_of_replicas = '0-all', column_policy = 'strict');
```

**Comment by joemoe** [comment]
thanks @tellezb for the clarification. we'll investigate what is happening here under the hood.

**Comment by seut** [comment]
@tellezb @benjaminrigaud 
We've found the bug and fixed it at our current master as well as at the 0.54 release branch.
It will be released with the next 0.54 hotfix release.
